src/server/registry.js

```diff
@@ -69,7 +96,7 @@ export class Registry {
     return { module: node.module, path, matchedPath: matchedParts.join("/") };
   }
 
-  endpoint(httpRequestMethod, url) {
+  endpoint(httpRequestMethod, url, openApiParameterDefinitions = []) {
```

Comment:
I don't like passing the raw parameter definitions in here because it couples Registry to the OpenAPI format. Instead we should pass in a data structure that's more to the point.

```ts
const parameterTypes = {
   path: {
     "petId": "number"
   },
   query: {
      category: "string"
   },
   headers: {}
}

```